app/src/main/java/com/example/taskyy/ui/viewmodels/AgendaViewModel.kt

```diff
@@ -124,23 +129,31 @@ class AgendaViewModel @Inject constructor(
 
     private fun setUserInitials() {
         val email = sharedPreferences.getString("email", "")
-        viewModelScope.launch {
-            val name = agendaRepository.getUserName(email!!)
-            sharedPreferences.edit().putString("name", name).apply()
-            state = state.copy(name = name)
-            state = state.copy(initials = name
-                .split(' ')
-                .mapNotNull { it.firstOrNull()?.toString() }
-                .reduce { acc, s -> acc + s })
+        if (savedStateHandle.get<String>("userInitials") != null) {
+            state = state.copy(initials = savedStateHandle.get<String>("userInitials") ?: "")
+        } else {
+            viewModelScope.launch {
+                val name = agendaRepository.getUserName(email!!)
+                sharedPreferences.edit().putString("name", name).apply()
+                state = state.copy(name = name)
+                state = state.copy(initials = name
+                    .split(' ')
+                    .mapNotNull { it.firstOrNull()?.toString() }
+                    .reduce { acc, s -> acc + s })
```

Comment:
If the user has a middle name, this will result in 3 initials being shown. In that case I'd expect that only the first and last name's initials will be shown